docs/core/logger.md

```diff
@@ -50,8 +50,6 @@ These settings will be used across all logs emitted:
 | ---------------------- | ---------------------------------------------------------------------------------------------------------------- | ------------------------------- | ------------------- | ------------------------------------------------------ | ------------------- | --------------------- |
 | **Service name**       | Sets the name of service of which the Lambda function is part of, that will be present across all log statements | `POWERTOOLS_SERVICE_NAME`       | `service_undefined` | Any string                                             | `serverlessAirline` | `serviceName`         |
 | **Logging level**      | Sets how verbose Logger should be, from the most verbose to the least verbose (no logs)                          | `POWERTOOLS_LOG_LEVEL`          | `INFO`              | `DEBUG`, `INFO`, `WARN`, `ERROR`, `CRITICAL`, `SILENT` | `ERROR`             | `logLevel`            |
-| **Log incoming event** | Whether to log or not the incoming event when using the decorator or middleware                                  | `POWERTOOLS_LOGGER_LOG_EVENT`   | `false`             | `true`, `false`                                        | `false`             | `logEvent`            |
-| **Debug log sampling** | Probability that a Lambda invocation will print all the log items regardless of the log level setting            | `POWERTOOLS_LOGGER_SAMPLE_RATE` | `0`                 | `0.0` to `1`                                           | `0.5`               | `sampleRateValue`     |
```

Comment:
Rather than removing the `sampleRateValue` from here I would instead change the wording above to remove the "required" mention entirely.

My reasoning here is that none of the settings is really required since they all have defaults and [they are all typed as such](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/logger/src/types/Logger.ts#L28-L36).

So to sum up I'd change the first sentence from what it is now to something along the lines of: `The logger accepts three settings.` or something similar.

Reply:
My goal was to align it with python docs and have a dedicated section for env variables. 

I think the confusion comes from the phrase 'or pass them in the constructor' and because we mix environment variables and constructor settings, which overlap with exceptions. So if we keep three ENVs that are also part of the constructor people might assume that the other constructor values are missing from the docs. 

We could add additional deep link to API docs to the constructor. What do you think?

Reply:
I think having a section to the env variables or linking to the existing one in the main page is great.

I was mainly arguing against the "required" wording which is really not true, and at the same time wanting to have a place to get started quickly.

As a user I want to read at the very top of the page how to get started using the library either because I don't know or remember it. Having all the applicable settings surfaced early is useful and we have evidence that customers are doing this (we had multiple reports about `logEvent` being out of place here).

I'm in favor of deep linking the API docs and making the wording clearer if needed, but I wouldn't remove the parameter from here.